std/experimental/color/rgb.d

```diff
@@ -0,0 +1,852 @@
+        return this;
+    }
+}
+
+unittest
+{
+    alias UnsignedRGB = RGB!("rgb", ubyte);
+    alias SignedRGBX = RGB!("rgbx", byte);
+    alias FloatRGBA = RGB!("rgba", float);
+
+    // test construction
+    static assert(UnsignedRGB("0x908000FF")  == UnsignedRGB(0x80,0,0xFF));
+    static assert(FloatRGBA("0x908000FF")    == FloatRGBA(float(0x80)/float(0xFF),0,1,float(0x90)/float(0xFF)));
+
+    // test operators
+    static assert(-SignedRGBX(1,2,3) == SignedRGBX(-1,-2,-3));
+    static assert(-FloatRGBA(1,2,3)  == FloatRGBA(-1,-2,-3));
+
+    static assert(UnsignedRGB(10,20,30)  + UnsignedRGB(4,5,6) == UnsignedRGB(14,25,36));
+    static assert(SignedRGBX(10,20,30)   + SignedRGBX(4,5,6)  == SignedRGBX(14,25,36));
+    static assert(FloatRGBA(10,20,30,40) + FloatRGBA(4,5,6,7) == FloatRGBA(14,25,36,47));
+
+    static assert(UnsignedRGB(10,20,30)  - UnsignedRGB(4,5,6) == UnsignedRGB(6,15,24));
+    static assert(SignedRGBX(10,20,30)   - SignedRGBX(4,5,6)  == SignedRGBX(6,15,24));
+    static assert(FloatRGBA(10,20,30,40) - FloatRGBA(4,5,6,7) == FloatRGBA(6,15,24,33));
+
+    static assert(UnsignedRGB(10,20,30)  * UnsignedRGB(0,1,2) == UnsignedRGB(0,20,60));
+    static assert(SignedRGBX(10,20,30)   * SignedRGBX(0,1,2)  == SignedRGBX(0,20,60));
+    static assert(FloatRGBA(10,20,30,40) * FloatRGBA(0,1,2,3) == FloatRGBA(0,20,60,120));
+
+    static assert(UnsignedRGB(10,20,30)  / UnsignedRGB(1,2,3) == UnsignedRGB(10,10,10));
+    static assert(SignedRGBX(10,20,30)   / SignedRGBX(1,2,3)  == SignedRGBX(10,10,10));
+    static assert(FloatRGBA(2,4,8,16)    / FloatRGBA(1,2,4,8) == FloatRGBA(2,2,2,2));
+
+    static assert(UnsignedRGB(10,20,30)  * 2 == UnsignedRGB(20,40,60));
+    static assert(SignedRGBX(10,20,30)   * 2 == SignedRGBX(20,40,60));
+    static assert(FloatRGBA(10,20,30,40) * 2 == FloatRGBA(20,40,60,80));
+
+    static assert(UnsignedRGB(10,20,30)  / 2 == UnsignedRGB(5,10,15));
+    static assert(SignedRGBX(10,20,30)   / 2 == SignedRGBX(5,10,15));
+    static assert(FloatRGBA(10,20,30,40) / 2 == FloatRGBA(5,10,15,20));
+
+    // test format conversions
+    static assert(UnsignedRGB(SignedRGBX(0x20,0x30,-10))           == UnsignedRGB(0x40,0x60,0));
+    static assert(UnsignedRGB(FloatRGBA(1,0.5,0,1))                == UnsignedRGB(0xFF,0x80,0));
+    static assert(UnsignedRGB(FloatRGBA(UnsignedRGB(0xFF,0x80,0))) == UnsignedRGB(0xFF,0x80,0));
+    static assert(FloatRGBA(UnsignedRGB(0xFF,0x80,0))              == FloatRGBA(1,float(0x80)/float(0xFF),0,0));
+    static assert(FloatRGBA(SignedRGBX(127,-127,-128))             == FloatRGBA(1,-1,-1,0));
+}
+
+
+// TODO: represent packed colors, eg R5G6B5, etc
+struct PackedColor(string format, RGBColorSpace colorSpace = RGBColorSpace.sRGB)
+{
+    // TODO: we'll try and fabricate the packing algorithm based on the format string
+    //...
+
+    void pack(C)(C color);
+    C unpack(C)();
+}
+
+
+// std.conv style to!RGB function
+To to(To, From)(From color) if(isRGB!To && isRGB!From)
```

Comment:
There's no single intermediary. XYZ is at the center of the colour space universe, but many conversions require to follow a path; ie, xyY -> HSV goes: xyY -> XYZ -> RGB -> HSV
The colour space landscape is a fairly awkward graph: http://www.codeproject.com/KB/graphics/613798/ConversionPrinciple_klein.png